src/mds/MDBalancer.cc

```diff
@@ -1134,7 +1134,9 @@ void MDBalancer::maybe_fragment(CDir *dir, bool hot)
 {
   // split/merge
   if (bal_fragment_dirs && bal_fragment_interval > 0 &&
-      dir->is_auth() && !dir->inode->is_base()) {  // not root/base (for now at least)
+      dir->is_auth() &&
+      !dir->inode->is_base() &&  // not root/mdsdir (for now at least)
+      !dir->inode->is_stray()) { // not straydir
```

Comment:
Why? What issue is this fixing? Please make a tracker ticket.

Reply:
The issue is trivial. Above code queue straydir (can not be fragmented) for split.  Just waste few cpu cycles

Reply:
Okay, I'm okay with this one not having a ticket but the other commit does.